README.md

```diff
@@ -1,31 +1,52 @@
 
-You can use our Python API like this.
+Below, we illustrate how a developer might construct such a minimal product search example using our Python API. 
 
 ```python
-import caramel
-keys = ["A", "B", "C", "D", "E", "F", "G", "H", "I", "J"]
-values = [0, 0, 0, 1, 2, 0, 1, 4, 0, 1]
+from carameldb import Caramel
+keys = ["red shoes", "red puma shoes", "black bags", "phone case", "shoes red"]
+values = [["A1", "A2", "A3"], ["A3", "A4", "A1"], ["A6", "A7", "A8"], ["A11", "A17", "A40"], ["A2", "A3", "A1"]]
 # Constructing
-csf = caramel.CSF(keys, values)
+caramel = Caramel(keys, values)
 # Querying
-csf.query("D")  # Returns 1
+caramel.query("red shoes")  # Returns ["A1", "A2", "A3"]
 # Saving
-csf.save("map.csf")
+caramel.save("map.caramel")
 # Loading
-csf = caramel.load("map.csf")
+caramel = Caramel.load("map.caramel")
 ```
 
+## Succinct Retrieval Methods
+
+Succinct retrieval is a beautiful research area because it lies at the nexus of theory and practice. The computer science community has proposed several solutions to this problem in recent years through a combination of clever algorithmic insights and innovative performance engineering efforts. These developments included Minimal Perfect Hash Tables (MPH), Compressed Static Functions (CSFs), and Bumped Ribbon Retrieval (BuRR). In the near future, we envision including all of these implementations, in addition to our own CARAMEL data structure, within the CaramelDB project as a one-stop shop for succinct retrieval in the Python ecosystem. 
+
+## What is CARAMEL?
+
+CARAMEL (Compressed Array Representation And More Efficient Lookups) is the centerpiece data structure in CaramelDB, achieving state-of-the-art performance in many practical settings, especially when the values to retrieve are multi-sets and not individual objects. The core algorithmic primitive behind CARAMEL is the compressed static function (CSF). For a solid introduction to CSFs, see the excellent series of papers by Sebastiano Vigna's group. We draw heavy inspiration from their [Java implementation in Sux4J](https://sux4j.di.unimi.it).
+
+## Why CaramelDB?
+
+### Algorithmic and Engineering Improvements
+
+To our knowledge, we are the first open-source succinct retrieval library to support arbitrary data types for keys and values, enabling more flexibility for applications. For example, the Sux4J CSF implementation only supports strings and integer types. Compared to Sux4J, we also offer about 25% faster construction, for CSFs. On the algorithmic front, we have optimal-size Bloom prefilters for situations when one value is very common. We also have value-permutation methods to preprocess 2D arrays into a more compressible format, as well as tricks to implement ragged arrays and other lookup primitives using CSFs. Our paper describes these tricks and applications.
```

Comment:
Nit: spurious comma - "25% faster construction, for CSFs" -> "25% faster construction for CSFs"

Reply:
Fixed